File: docs/src/test/java/org/springframework/security/docs/features/authentication/authenticationpasswordstoragebcrypt/BCryptPasswordEncoderUsage.java
package org.springframework.security.docs.features.authentication.authenticationpasswordstoragebcrypt;

import static org.junit.Assert.assertTrue;

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

public class BCryptPasswordEncoderUsage {
	public void testBCryptPasswordEncoder() {
		// tag::bcryptPasswordEncoder[]
		// Create an encoder with strength 16
		BCryptPasswordEncoder encoder = new BCryptPasswordEncoder(16);
		String result = encoder.encode("myPassword");
		assertTrue(encoder.matches("myPassword", result));
		// end::bcryptPasswordEncoder[]
	}
}

File: docs/src/test/kotlin/org/springframework/security/kt/docs/features/authentication/authenticationpasswordstoragebcrypt/BCryptPasswordEncoderUsage.kt
package org.springframework.security.kt.docs.features.authentication.authenticationpasswordstoragebcrypt

import org.junit.Assert.assertTrue
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder

class BCryptPasswordEncoderUsage {
    fun testBCryptPasswordEncoder() {
        // tag::bcryptPasswordEncoder[]
        // Create an encoder with strength 16
        val encoder = BCryptPasswordEncoder(16)
        val result: String = encoder.encode("myPassword")
        assertTrue(encoder.matches("myPassword", result))
        // end::bcryptPasswordEncoder[]
    }
}
